.cicd/docker/ubuntu-18.04.dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM ubuntu:18.04
+# install dependencies
+RUN apt-get update && \
+    apt-get install -y build-essential git automake python2.7 python2.7-dev python3 python3-dev curl ccache
+# build cmake
+RUN curl -LO https://cmake.org/files/v3.13/cmake-3.13.2.tar.gz && \
+    tar -xzf cmake-3.13.2.tar.gz && \
+    cd cmake-3.13.2 && \
+    ./bootstrap --prefix=/usr/local && \
+    make -j $(nproc) && \
+    make install && \
+    rm -f /cmake-3.13.2.tar.gz && rm -rf /cmake-3.13.2
+# build clang
+RUN git clone --single-branch --branch release_80 https://git.llvm.org/git/llvm.git clang8 && cd clang8 && git checkout 18e41dc && \
```

Comment:
This block is a little confusing and probably hard to update with all of the separate commit hashes for the submodules.  Is there a really high cost for just cloning the branch and doing the submodule update?